Annotato/Annotato/WebSocket/WebSocketManager.swift

```diff
@@ -2,26 +2,17 @@ import Foundation
 import AnnotatoSharedLibrary
 
 class WebSocketManager: ObservableObject {
-    static let shared = WebSocketManager()
-
     private(set) var socket: URLSessionWebSocketTask?
 
     @Published private(set) var message: Data?
 
-    private init() { }
-
-    func setUpSocket() {
-        guard let userId = AnnotatoAuth().currentUser?.uid else {
-            AnnotatoLogger.error("Unable to retrieve user id.", context: "WebSocketManager::setUpSocket")
-            return
-        }
-
-        guard let url = URL(string: "\(RemotePersistenceService.baseWsAPIUrl)/ws/\(userId)") else {
+    func setUpSocket(urlString: String) {
```

Comment:
This is not possible. We cannot store the URL in a init function because the user id might not be available before login since we store the websocket in the scene delegate.

Reply:
Then can we
1. set the URL instance property once the user id is available
2. init with a callback that will produce the url value? i.e. `init(urlProducer: () -> URL?)` then in `setUpSocket` we can call this callback to get the URL

I think in both cases it's still cleaner than having to pass the url string each time we want to set up the socket? And also prevents one socket from being reused for another endpoint. If you disagree then we can just go with the current implementation too.